File: englising-be/src/main/java/org/englising/com/englisingbe/music/entity/Artist.java
package org.englising.com.englisingbe.music.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "artist")
public class Artist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "artist_id")
    private Long artistId;

    @Column(name = "name")
    private String name;

    @Column(name = "genres")
    private String genres;

    @Column(name = "spotify_id")
    private String spotifyId;

    @Column(name = "spotify_popularity")
    private Integer spotifyPopularity;

    @Column(name = "image")
    private String coverImage;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}

File: englising-data-youtube-crawler/model/artist.py
from sqlalchemy import Column, ForeignKey, Integer, String, Float, Boolean, TIMESTAMP, DECIMAL, Text, Date, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from model.base import Base


class Artist(Base):
    __tablename__ = 'artist'
    artist_id = Column(Integer, primary_key=True)
    name = Column(String(255), nullable=False)
    genres = Column(String(255), nullable=True)
    spotify_id = Column(String(255), nullable=False, unique=True)
    spotify_popularity = Column(Integer, nullable=True)
    image = Column(String(512))
    created_at = Column(TIMESTAMP, nullable=False, default=func.now())
    updated_at = Column(TIMESTAMP, onupdate=func.now())

    tracks = relationship('Track', secondary='artist_track', back_populates='artists')

